code/components/citizen-server-impl/src/NoticeLogicProcessor.cpp

```diff
@@ -0,0 +1,140 @@
+#include <StdInc.h>
+#include "NoticeLogicProcessor.h"
+
+NoticeLogicProcessor::NoticeLogicProcessor(fx::ServerInstanceBase* server)
+{
+	m_cvManager = server->GetComponent<console::Context>()->GetVariableManager();
+	m_resManager = server->GetComponent<fx::ResourceManager>()->GetCurrent();
+}
+
+bool NoticeLogicProcessor::ProcessNoticeRule(json& ruleRef, uint32_t nestingLevel)
+{
+	if (nestingLevel >= 10)
+		throw std::invalid_argument("Maximum nesting level for notice rules was exceeded");
+	if (!ruleRef.is_object())
+		return false;
```

Comment:
Style guide: use braces everywhere and group them by functionality :)
For some instances down below; you can use ternary operators or return the condition check straight away

Reply:
Added the braces and made use of ternary operators where applicable 👍